lib/usd/ui/views/USDImportDialog.cpp

```diff
@@ -0,0 +1,225 @@
+//
+// Copyright 2019 Autodesk
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include "USDImportDialog.h"
+#include "factories/TreeModelFactory.h"
+//#include "TreeHeader.h"
+#include "ItemDelegate.h"
+#include "ui_USDImportDialog.h"
+
+#include <stdexcept>
+
+MAYAUSD_NS_DEF {
+
+IUSDImportView::~IUSDImportView() { }
+
+USDImportDialog::USDImportDialog(const std::string& filename, QWidget* parent)
+	: QDialog{ parent }
+	, fUI{ new Ui::ImportDialog() }
+	, fStage{ UsdStage::Open(filename, UsdStage::InitialLoadSet::LoadNone) }
+	, fFilename{ filename }
+	, fRootPrimPath("/")
+{
+	if (!fStage)
+		throw std::invalid_argument("Invalid filename passed to USD Import Dialog");
+
+	fUI->setupUi(this);
+
+	// WIP - the search buttons are not implemented yet, so disable them.
+	fUI->findPrev->setHidden(true);
+	fUI->findNext->setHidden(true);
+	fUI->findAll->setHidden(true);
```

Comment:
WIP - I hid these buttons for now as they aren't implemented yet.

Reply:
I think it would be preferable for work-in-progress functionality to **not** be included in a PR. When the functionality is ready, a PR dedicated to adding the new buttons and the code that drives them becomes easier to review, as does this PR.